OndaEDFSchemas.jl/test/runtests.jl

```diff
@@ -7,61 +7,73 @@ using StableRNGs
-                physical_maximum=2.0,
-                digital_minimum=-1f4,
-                digital_maximum=1f4,
-                prefilter="HP 0.1Hz; LP 80Hz; N 60Hz",
-                samples_per_record=128,
-                seconds_per_record=1.0,
-                kind=ingested ? "eeg" : missing,
-                channel=ingested ? "cz-m1" : missing,
-                sample_unit=ingested ? "microvolt" : missing,
-                sample_resolution_in_unit=ingested ? 1f-4 : missing,
-                sample_offset_in_unit=ingested ? 1.0 : missing,
-                sample_type=ingested ? "float32" : missing,
-                sample_rate=ingested ? 1/128 : missing,
-                # errors, use `nothing` to indicate no error
-                error=errored ? "Error blah blah" : nothing)
+    PlanVersion = v == 1 ? PlanV1 : PlanV2
+    return PlanVersion(; label="EEG CZ-M1",
+                       transducer_type="Ag-Cl electrode",
+                       physical_dimension="uV",
+                       physical_minimum=0.0,
+                       physical_maximum=2.0,
+                       digital_minimum=-1f4,
+                       digital_maximum=1f4,
+                       prefilter="HP 0.1Hz; LP 80Hz; N 60Hz",
+                       samples_per_record=128,
+                       seconds_per_record=1.0,
+                       channel=ingested ? "cz-m1" : missing,
+                       sample_unit=ingested ? "microvolt" : missing,
+                       sample_resolution_in_unit=ingested ? 1f-4 : missing,
+                       sample_offset_in_unit=ingested ? 1.0 : missing,
+                       sample_type=ingested ? "float32" : missing,
+                       sample_rate=ingested ? 1/128 : missing,
+                       error=errored ? "Error blah blah" : nothing,
+                       specific_kwargs...)
 end
 
-function mock_file_plan(n; rng=GLOBAL_RNG)
-    tbl = [mock_file_plan(; rng) for _ in 1:n]
+function mock_file_plan(n; v, rng=GLOBAL_RNG)
+    tbl = [mock_file_plan(; v, rng) for _ in 1:n]
     return tbl
 end
 
-function mock_file_plan(; rng=GLOBAL_RNG)
-    plan = mock_plan(; rng)
-    return FilePlan(Tables.rowmerge(plan;
-                                    edf_signal_index=rand(rng, Int),
-                                    onda_signal_index=rand(rng, Int)))
+function mock_file_plan(; v, rng=GLOBAL_RNG)
+    plan = mock_plan(; v, rng)
+    PlanVersion = v == 1 ? FilePlanV1 : FilePlanV2
+    return PlanVersion(Tables.rowmerge(plan;
+                                       edf_signal_index=rand(rng, Int),
+                                       onda_signal_index=rand(rng, Int)))
 end
 
-@testset "ondaedf.plan@1" begin
-    rng = StableRNG(10)
-    plans = mock_plan(30; rng)
-    @test nothing === Legolas.validate(plans, Legolas.Schema("ondaedf.plan@1"))
-    tbl = Arrow.Table(Arrow.tobuffer(plans))
-    @test isequal(Tables.columntable(tbl), Tables.columntable(plans))
-end
+for v in [1, 2]
```

Comment:
you _could_ do this as a testset if you wanted...

```suggestion
@testset "Schema version $v" for v in (1, 2)
```